### PR TITLE
Replace all usages of findProperty

### PR DESCRIPTION
**Pull request by ansman**
It is not compatible with project isolation, `providers.gradleProperty` is however.

This PR also change all build script related usages, though this isn't needed for consumers of course.

### CODE REVIEW

**Review by ZacSweers — APPROVED** [review]
Thanks!

**Comment by ZacSweers** [comment]
@ansman looks like CI is failing

**Comment by ansman** [comment]
Fill fix tomorrow, not in front of a computer today

File: build.gradle.kts [hunk 1]
```diff
@@ -77,7 +77,8 @@ subprojects {
   pluginManager.withPlugin("java") {
     // javaReleaseVersion can be set to override the global version
     val jvmTargetProvider =
-      provider<String> { findProperty("moshix.javaReleaseVersion") as? String? }
+      providers
+        .gradleProperty("moshix.javaReleaseVersion")
```
Comment (ZacSweers) [inline build.gradle.kts:81]: The problem is here unfortunately, `providers.gradleProperty()` doesn't work for subprojects, only root projects

**Pull request merged**